tests/chainer_tests/functions_tests/activation_tests/test_leaky_relu.py

```diff
@@ -47,56 +41,20 @@ def setUp(self):
             self.check_backward_options = {'atol': 5e-4, 'rtol': 5e-3}
             self.check_double_backward_options = {'atol': 5e-3, 'rtol': 5e-2}
 
-    def check_forward(self, x_data, backend_config):
-        if backend_config.use_cuda:
-            x_data = cuda.to_gpu(x_data)
-        x = chainer.Variable(x_data)
-
-        with backend_config:
-            y = functions.leaky_relu(x, slope=self.slope)
-        self.assertEqual(y.data.dtype, self.dtype)
-
-        expected = numpy.where(self.x >= 0, self.x, self.x * self.slope)
-
-        testing.assert_allclose(
-            expected, y.data, **self.check_forward_options)
-
-    def test_forward(self, backend_config):
-        self.check_forward(self.x, backend_config)
-
-    def check_backward(self, x_data, y_grad, backend_config):
-        if backend_config.use_cuda:
-            x_data = cuda.to_gpu(x_data)
-            y_grad = cuda.to_gpu(y_grad)
-
-        def f(x):
-            return functions.leaky_relu(x, self.slope)
-
-        with backend_config:
-            gradient_check.check_backward(
-                f, x_data, y_grad, dtype=numpy.float64,
-                **self.check_backward_options)
-
-    def test_backward(self, backend_config):
-        self.check_backward(self.x, self.gy, backend_config)
-
-    def check_double_backward(self, x_data, y_grad,
-                              x_grad_grad, backend_config):
-        if backend_config.use_cuda:
-            x_data = cuda.to_gpu(x_data)
-            y_grad = cuda.to_gpu(y_grad)
-            x_grad_grad = cuda.to_gpu(x_grad_grad)
-
-        def f(x):
-            return functions.leaky_relu(x, self.slope)
+    def generate_inputs(self):
+        x = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        if self.slope == 'random':
```

Comment:
To keep similar terms together, better to leave these `self.slope` lines in setUp.